Content.Server/CartridgeLoader/Cartridges/MessagesCartridgeSystem.cs

```diff
@@ -0,0 +1,310 @@
+
+        if (component.UserName == null)
+        {
+            if ((component.ConnectedId != null) && (TryComp(component.ConnectedId, out IdCardComponent? idCardComponent)))
+            {
+                if (idCardComponent.FullName != "")
+                {
+                    component.UserName = idCardComponent.FullName+"("+idCardComponent.JobTitle+")";
+                }
+                else
+                {
+                    component.UserName = "Unknown "+"("+idCardComponent.JobTitle+")";
+                }
+            }
+            if (component.UserUid != null && component.UserName != null)
+                component.NameDict[component.UserUid] = component.UserName;
+        }
+
+        //_sawmill.Debug("Updating cart");
+        if ((component.MessagesQueue.Count > 0) || component.DeadConnection)
+        {
+            if (!(GetActiveServer(component, mapId) is var (serverUid, messageServer)))
+            {
+                component.DeadConnection=true;
+                return;
+            }
+            else
+            {
+                if (component.DeadConnection)
+                    PullFromServer(uid, component, messageServer);
+                component.DeadConnection=false;
+            }
+
+
+            var tempMessageQueue = new List<MessagesMessageData>(component.MessagesQueue);
+            foreach (var message in tempMessageQueue)
+            {
+                _messagesServerSystem.PdaToServerMessage(serverUid, messageServer,message);
+                component.MessagesQueue.Remove(message);
+                component.Messages.Add(message);
+                UpdateUiState(uid, Transform(uid).ParentUid, component);
+            }
+        }
+    }
+
+    public (EntityUid, MessagesServerComponent)? GetActiveServer(MessagesCartridgeComponent component,MapId mapId)
+    {
+       // _sawmill.Debug("Looking for server");
+        var servers = EntityQueryEnumerator<MessagesServerComponent, ApcPowerReceiverComponent, TransformComponent>();
+        while(servers.MoveNext(out var uid, out var messageServer, out var power, out var transform))
+        {
+            if (messageServer.EncryptionKey != component.EncryptionKey)
+                continue;
+
+            if (transform.MapID == mapId &&
+                power.Powered)
+            {
+                //_sawmill.Debug("Server found");
+                return (uid, messageServer);
+            }
+        }
+       // _sawmill.Debug("No server");
+        return null;
+    }
+
+    private string GetName(MessagesCartridgeComponent component, string key)
+    {
+        if (!(component.NameDict.ContainsKey(key)))
+        {
+            return "Unknown";
+        }
+        else
+        {
+            if (component.NameDict[key][0] == '(')
+            {
+                return "Unknown "+component.NameDict[key];
+            }
+            else
+            {
+                return component.NameDict[key];
+            }
+        }
+    }
+
```

Comment:
As above. To be honest just use identity to get the name at runtime instead of having to fix this properly.